atomic_reactor/plugins/pre_resolve_composes.py

```diff
@@ -110,8 +111,18 @@ def run(self):
             self.wait_for_composes()
         except WaitComposeToFinishTimeout as e:
             self.log.info(str(e))
-            self.log.info('Canceling the compose %s', e.compose_id)
-            self.odcs_client.cancel_compose(e.compose_id)
+            self.log.info('The compose %s timed out', e.compose_id)
+
+            preserve_composes = set(self.compose_ids).union(self.parent_compose_ids)
+            cancel_composes = set(self.all_compose_ids) - preserve_composes
+            if cancel_composes:
+                self.log.info('Canceling unfinished composes which were created by the build: %s',
+                              cancel_composes)
+
```

Comment:
For others: this part has been reworked to prevent canceling compose not triggered by OSBS. Please re-review